datafusion/expr/src/signature.rs

```diff
@@ -145,6 +145,9 @@ pub enum ArrayFunctionSignature {
     /// The function takes a single argument that must be a List/LargeList/FixedSizeList
     /// or something that can be coerced to one of those types.
     Array,
+    /// Specialized Signature for MapArray
+    /// The function takes a single argument that must be a MapArray
+    MapArray,
```

Comment:
introduce `MapArray`  in the function signature

Reply:
👍 

I wonder if this is another example of where `LogicalType`s proposed by @notfillipo in https://github.com/apache/datafusion/issues/11513 would be useful

Reply:
Yes, I think so. The signature will benefit from that decouple draft and make the implementation more concise and clear.